<launch>

  <arg name="robot_ip" />
  
  <arg name="robot_dof" />
  <!-- load the default move_group planner (not xarm_simple_planner) -->
  <arg name="show_rviz" default="true" />
  <!-- NO_GUI_CTRL means that Rviz configuration is just for display purpose, no (Moveit) control panel loaded -->
  <arg name="no_gui_plan" default="true" />
  <arg name="add_gripper" default="false" />
  <arg name="add_vacuum_gripper" default="false" />
  <arg name="report_type" default="normal" />
  <arg name="ext_ns" default="" />
  <arg name="robot_type" default="xarm"/>
  <arg name="model1300" default="false" />

  <remap from="move_group/display_planned_path" to="$(arg ext_ns)/move_group/display_planned_path" />
  <include file="$(find xarm_planner)/launch/moveit_real_arm_configurations.launch">
    <arg name="arm_dof" value="$(arg robot_dof)" />
    <arg name="robot_ip" value="$(arg robot_ip)" />
    <arg name="end_effector" value="_gripper" if="$(arg add_gripper)" />
    <arg name="end_effector" value="_vacuum_gripper" if="$(arg add_vacuum_gripper)" />
    <arg name="report_type" value="$(arg report_type)" />
    <arg name="ext_ns" value="$(arg ext_ns)" />
    <arg name="robot_type" value="$(arg robot_type)" />
    <arg name="model1300" value="$(arg model1300)" />
    <arg name="no_gui_plan" value="$(arg no_gui_plan)" />
  </include>

  <param name="robot_name" type="str" value = "$(eval arg('robot_type') if arg('robot_type') == 'uf850' else arg('robot_type') + str(arg('robot_dof')))"/>
  <!-- bring up the simple planner (based on move_group) that provide plan service and execution server -->
  <node name="xarm_move_group_planner" pkg="xarm_planner" type="xarm_simple_planner" />

</launch>

